napari/utils/events/event.py

```diff
@@ -299,6 +299,8 @@ def __init__(
         # count number of times this emitter is blocked for each callback.
         self._blocked: Dict[Optional[Callback], int] = {None: 0}
         self._block_counter: _WeakCounter[Optional[Callback]] = _WeakCounter()
+        self._delay_semaphore = 0
+        self._delay_last_event = None
```

Comment:
Typing info is a bonus

```suggestion
        self._delay_semaphore: int = 0
        self._delay_last_event: Optional[Event] = None
```